integration_tests/test_helper.go

```diff
@@ -0,0 +1,348 @@
+package integrationtests
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/facebookgo/symwalk"
+	svchost "github.com/hashicorp/terraform-svchost"
+	"github.com/hashicorp/terraform/addrs"
+	terraspec "github.com/nhurel/terraspec/lib"
+)
+
+// GetTerraspec computes the path of the terraspec executable.
+// If it cannot be found it is build.
+func GetTerraspec(t *testing.T, rootDir string) string {
+	terraspecFileName := "terraspec"
+	if runtime.GOOS == "windows" {
+		terraspecFileName = terraspecFileName + ".exe"
+	}
+	terraspecPath := rootDir + "/" + terraspecFileName
+
+	_, err := os.Stat(terraspecPath)
+	if os.IsNotExist(err) {
+		changeBack := Chdir(t, rootDir)
+		defer changeBack()
+
+		buildTerraspec(t, terraspecFileName)
+	} else if err != nil {
+		t.Fatalf("Could not stat file %s: %v", terraspecPath, err)
+	}
+
+	return terraspecPath
+}
+
+func buildTerraspec(t *testing.T, terraspecFileName string) {
+	t.Logf("Building terraspec executable: go build -o %s .", terraspecFileName)
+	cmd := exec.Command("go", "build", "-o", terraspecFileName, ".")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("Could not build terraspec, output is\r\n%s\r\nerror: %v", string(output), err)
+	}
+}
+
+// RunTerraspec runs the given terraspec executable in the specified project path.
+func RunTerraspec(t *testing.T, terraspecPath string, projectPath string) {
+	changeBack := Chdir(t, projectPath)
+	defer changeBack()
+
+	cwd := Getwd(t)
+
+	t.Logf("Running %s in %s", terraspecPath, cwd)
+	output, err := exec.Command(terraspecPath).CombinedOutput()
+	t.Logf("Terraspec output is:\r\n%s", string(output))
+	if err != nil {
+		t.Fatalf("Error while executing terraspec: %v", err)
+	}
+}
+
+
+
+// GetTerraform downloads terraform with the stated version into the root project directory.
+func GetTerraform(t *testing.T, version string, rootDir string) (string, func()) {
+	terraformZIPFileName := "terraform"
+	terraformFileName := "terraform_v" + version
+	if runtime.GOOS == "windows" {
+		terraformZIPFileName += ".exe"
+		terraformFileName += ".exe"
+	}
+
+	terraformPath := rootDir + "/" + terraformFileName
+	_, err := os.Stat(terraformPath)
+	if err == nil {
+		return terraformPath, func() {}
+	}
+
+	zipFilePath := "terraform.zip"
+	terraformURL := fmt.Sprintf("https://releases.hashicorp.com/terraform/%s/terraform_%s_%s_%s.zip", version, version, runtime.GOOS, runtime.GOARCH)
+	downloadFile(t, zipFilePath, terraformURL)
+	defer os.Remove(zipFilePath)
+
+	terraformPath, err = unzip(zipFilePath, terraformZIPFileName, terraformFileName, rootDir)
+	if err != nil {
+		t.Fatalf("Could not unzip terraform download: %v", err)
+	}
+
+	return terraformPath, func() {}
```

Comment:
the returned deleteTerraform function is alway empty ?

Reply:
I dont delete it anymore because I kind of "cache" them for all the tests. I will remove that return param.